qringbuf.go

```diff
@@ -1,3 +1,168 @@
+// Package qringbuf provides a concurrency-friendly zero-copy abstraction of
+// io.ReadAtLeast() over a pre-allocated ring-buffer, populated asynchronously
+// by a standalone goroutine. It is primarily designed for processing a series
+// of arbitrary streams each comprised of variable-length records.
+//
+// Specifically an object of this package makes the following guarantees:
+//  • Memory is allocated only at construction time, never during streaming
+//  • StartFill() spawns off a single goroutine (collector) which terminates when:
+//    ◦ It reaches readLimit, if one was supplied
+//    ◦ It receives any error from the wrapped reader (including io.EOF)
+//    ◦ It had a chance to evaluate a signal received from StopFill()
+//      (most io.Read() operations are not cancelable, and may block forever)
+//  • Every call to NextRegion() blocks until it can return:
+//    ◦ A *Region object representing a contiguous slice of at least MinRegion bytes
+//    ◦ A smaller "trailing" *Region and an io.EOF or ErrCollectorStopped
```

Comment:
I'm not sure what a "trailing" region means here. It might make sense to give a small example of the methods on the API and how they're typically used before the guarantees section, to give the reader context

Reply:
See https://github.com/ipfs/go-qringbuf/pull/6/commits/4d65af211de